daml-foundations/daml-ghc/daml-compiler/src/DA/Service/Daml/Compiler/Impl/Dar.hs

```diff
@@ -102,10 +105,10 @@ buildDar service file mbExposedModules pkgName sdkVersion buildDataFiles dalfInp
               show (S.toList missingExposed)
       let dalf = encodeArchiveLazy pkg
       -- get all dalf dependencies.
-      deps <- getDalfDependencies file
-      dalfDependencies<- forM deps $ \(DalfDependency depName fp) -> do
-        pkgDalf <- liftIO $ BS.readFile fp
-        return (depName, pkgDalf)
+      dalfDependencies <-
+          fmap
+              (map (\(unitId, pkg) -> (T.pack $ unitIdString unitId, dalfPackageBytes pkg)) . Map.toList)
+              (getDalfDependencies file)
```

Comment:
That might be easier to read as
```haskell
dalfDependencies0 <- getDalfDependencies file
let dalfDependencies = [ ... | (unitId, pkg) <- Map.toList dalfDependencies0 ]
```
(I with `let` was not recursive by default and we could drop the `0` suffix...)